src/pages/dar_application/DataUseAgreements.jsx

```diff
@@ -14,11 +13,77 @@ export default function DataUseAgreements(props) {
     darCode,
     isAttested,
     cancelAttest,
+    datasets
   } = props;
+  const [daas, setDaas] = useState([]);
+
+  useEffect(() => {
+    const init = async () => {
+      try {
+        const daaList = await DAA.getDaas();
```

Comment:
This means we're getting ALL data access agreements every time we load this page. I think a more sustainable approach would be to get a list of DAAs for a distinct set of dataset ids OR a list of DAAs for a DAR reference id (I like this approach better) and that way we can move the filtering logic to the server which should allow the UI to be more streamlined. This would be a new API/ticket and an incremental improvement.

A related issue we have here is that there is the case where there are no DAAs for a particular dataset. We should clarify what the requirements should be in that case which can leave a DAR in a strange state.

Reply:
I think getting a list of DAAs for a DAR reference id makes sense, the ticket to add that functionality is linked here: https://broadworkbench.atlassian.net/browse/DCJ-364

I don't think there will be a case where there are no DAAs for a particular dataset. If a dataset is associated with a DAC, it will have a DAA associated with it. In the case that a dataset is Open Access or External to DUOS, we should not be creating a DAR for it. The ticket for addressing this existing bug allowing us to create DARs that are Open Access or External to DUOS is linked here: https://broadworkbench.atlassian.net/browse/DCJ-359